driver/bpf/fillers.h

```diff
@@ -7319,4 +7319,75 @@ FILLER(sys_newfstatat_x, true)
 	uint32_t flags = bpf_syscall_get_argument(data, 3);
 	return bpf_push_u32_to_ring(data, newfstatat_flags_to_scap(flags));
 }
+
+
+FILLER(sys_process_vm_readv_x, true)
+{
+	const struct iovec __user *iov;
+	unsigned long iovcnt;
+
+	/* Parameter 1: ret (type: PT_INT64) */
+	long retval = bpf_syscall_get_retval(data->ctx);
+	int res = bpf_push_s64_to_ring(data, (int32_t)retval);
+	CHECK_RES(res);
+
+	/* Parameter 2: pid (type: PT_PID) */
+	pid_t pid = (int32_t)bpf_syscall_get_argument(data, 0);
+	res = bpf_push_s64_to_ring(data, (int64_t)pid);
+	CHECK_RES(res);
+
+	/* Parameter 3: data (type: PT_BYTEBUF) */
+	if (retval > 0)
+	{
+		iov = (const struct iovec __user *)bpf_syscall_get_argument(data, 3);
```

Comment:
Question/Doubt -- IIRC args 3 and 4 point to `remote_iov` and `riovcnt`, which from my understanding is the memory area of the PID we're reading from. Since we do it only in case of successful read, wouldn't it make more sense to read from the `local_iov` instead (which is the destination buffer)? I also don't know what are the "locking" guarantees the kernel provides for the remote process we're reading from, but at least we know that the local thread's execution will be block up until we exit from this filler. I know this is still not optimal, but maybe the best tradeoff IMO.

In turn, this also applies to `process_vm_writev_x`. Let me know what you think.

cc @FedeDP and @Andreagit97 for opinions

ref: https://man7.org/linux/man-pages/man2/process_vm_readv.2.html

Reply:
I 100% second this; great catch!